### PR TITLE
fix(scripts): run package scripts through pnpm's shell emulator

### LINKED ISSUE #733: pnpm docs:build fails on Windows because docs scripts use POSIX inline environment syntax
## Summary

`pnpm docs:build` fails on Windows before the docs site is built. Two scripts in
`docs/package.json` set environment variables with POSIX inline syntax, which `cmd.exe` does not
understand:

```
"prisma:generate": "DATABASE_URL=${DATABASE_URL:-postgresql://user:password@localhost:5432/farmjs} node ../scripts/run-prisma-generate.mjs",
"build": "... && pnpm run prisma:generate && NODE_OPTIONS=--max-old-space-size=6144 farm build"
```

`${DATABASE_URL:-default}` is shell parameter expansion and `VAR=value command` is a shell
prefix. pnpm runs scripts through `cmd.exe` on Windows by default, so the first token is treated
as a program name.

## Reproduction

On Windows, from the repository root:

```
pnpm docs:build
```

```
> DATABASE_URL=${DATABASE_URL:-postgresql://user:password@localhost:5432/farmjs} node ../scripts/run-prisma-generate.mjs
'DATABASE_URL' is not recognized as an internal or external command,
operable program or batch file.
 ELIFECYCLE  Command failed with exit code 1.
```

With `DATABASE_URL` exported beforehand and the remaining steps run by hand, the docs build
completes, so only the script syntax is at fault.

## Impact

Windows contributors cannot run the docs build that pull requests are expected to verify.
`scripts/run-prisma-generate.mjs` already exists to handle a Windows specific `prisma generate`
race, so Windows is a supported environment for this step; the script is just unreachable from a
Windows shell.

`release:build` in the root `package.json` uses the same `NODE_OPTIONS=... pnpm build` form.

## Suggested direction

Either of these keeps the scripts as they are for POSIX users:

- Add `shell-emulator=true` to `.npmrc`. pnpm then interprets `VAR=value` prefixes and
  `${VAR:-default}` on every platform.
- Move the default into the Node script with `process.env.DATABASE_URL ??= "..."`, and pass the
  heap size through the script rather than a `NODE_OPTIONS` prefix.

Happy to send a PR for either.

### PR DESCRIPTION
**Pull request by kira-1011**
## Problem

`pnpm docs:build` fails on Windows before the docs site is built:

```
> DATABASE_URL=${DATABASE_URL:-postgresql://user:password@localhost:5432/farmjs} node ../scripts/run-prisma-generate.mjs
'DATABASE_URL' is not recognized as an internal or external command
```

Fixes #733.

## Root cause

pnpm runs package scripts through `cmd.exe` on Windows. `VAR=value command` is a POSIX shell
prefix and `${VAR:-default}` is shell parameter expansion, so `cmd.exe` treats `DATABASE_URL=...`
as a program name. The docs `build` script has a second instance in `NODE_OPTIONS=... farm build`,
and the same form appears in `release:build`, the deployment-preset examples and the benchmarks.

## Change

Add `shell-emulator=true` to a repository `.npmrc`. pnpm then runs scripts through its JavaScript
shell emulator on every platform, which interprets variable prefixes and `${VAR:-default}`
expansion, so no script needs rewriting.

A regression test in `scripts/pnpm-script-shell.test.js` runs a probe script through pnpm inside a
throwaway package carrying the repository's `.npmrc`, and checks that an inline prefix reaches the
command and that a default expansion falls back when unset and keeps an explicit value. It is added
to the root `test` script.

## Safety and compatibility

The emulator applies on every platform, not only Windows. Tracked package scripts use `&&` chains,
variable prefixes and `test/*.test.mjs` globs, all of which run unchanged under it; the CLI, sentry
and core scripts were run to confirm. One semantic difference from bash: `${VAR:-default}` applies
the default when the variable is unset, not when it is set to an empty string. The repository
relies on the unset case, and the test documents the distinction.

## Verification

- `pnpm docs:build` on Windows (Prisma Client generated and the build completes; it failed before
  reaching `farm build` without the change)
- `node --test scripts/pnpm-script-shell.test.js` (3 tests passed; all 3 fail with `.npmrc` removed)
- `node --test scripts/pack-public-packages.test.js scripts/release-beta.test.js scripts/publish-beta.test.js scripts/run-prisma-generate.test.js scripts/pnpm-script-shell.test.js` (17 tests passed)
- `pnpm --filter @farm.js/cli test` (88 tests passed, exercises the `test/*.test.mjs` glob)
- `pnpm --filter @farm.js/sentry test` (47 tests passed)
- `pnpm --filter @farm.js/core type-check`
- `pnpm lint` at the root (0 errors)
- `pnpm docs:check-navigation`
- `git diff --check`

## Documentation and release

No documentation, release metadata or generated artifacts changed.

---
## Summary by cubic
Fixes `pnpm docs:build` failing on Windows, where `cmd.exe` can't run `VAR=value` prefixes or `${VAR:-default}` expansions used in package scripts. Setting `shell-emulator=true` in `.npmrc` makes pnpm run scripts with its own shell emulator on every platform.

**Bug Fixes**
- Adds a regression test in `scripts/pnpm-script-shell.test.js` that verifies prefixes and `${VAR:-default}` expansion semantics; it's now part of the root `test` script.
- The emulator differs from bash in one edge case: the default applies only when the variable is unset, not when empty, a behavior the repo relies on and the new tests pin.

<sup>Written for commit b8266add99b0086a9b722820684f7e7fe95d43d2. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/farming-labs/farm.js/pull/734?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Scripts across the repo set variables with `VAR=value command` and
+# `${VAR:-default}`, which cmd.exe cannot run. The emulator interprets that
+# syntax the same way on every platform, so `pnpm docs:build` and the
+# deployment-preset examples work on Windows without rewriting each script.
+shell-emulator=true
```
[the remark this answers is not in the document]
Comment (kira-1011) [reply .npmrc:5]: Checked against the pinned version rather than the changelog. A throwaway package with this `.npmrc` and a `VAR=${VAR:-fallback} node ...` script, run with `npx pnpm@8.12.1 run probe`, prints `fallback`, and 8.12.1 emits no unsupported-setting warning. The `windows-latest` job on this PR, which uses the pinned version through `pnpm/action-setup`, also passes `scripts/pnpm-script-shell.test.js`. The setting predates 8.15; no bump is needed.

[the remark this answers is not in the document]
Comment (kira-1011) [reply .npmrc:5]: Agreed that the scope is every platform, which is why the check went beyond the docs script. Every tracked `package.json` script in the repository uses `&&` chains, `VAR=value` prefixes and `test/*.test.mjs` globs, all of which the emulator runs; the CLI, sentry and core scripts were run under it locally. CI on this PR installs and tests on ubuntu and windows with the setting active, which exercises dependency lifecycle scripts as well. The one semantic difference found, `${VAR:-default}` defaulting only when unset, is documented and now pinned by a test.

File: scripts/pnpm-script-shell.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const result = runScript(`PROBE_VAR=inline ${PRINT}`);
+
+  assert.equal(result.status, 0, result.stderr);
+  assert.equal(result.stdout, "inline");
+});
+
+test("a ${VAR:-default} expansion falls back when the variable is unset", () => {
+  // The emulator applies the default for an unset variable only; unlike bash it
+  // leaves an empty string alone. The repository relies on the unset case.
+  const result = runScript(`PROBE_VAR=\${PROBE_VAR:-fallback} ${PRINT}`, { PROBE_VAR: undefined });
```
[the remark this answers is not in the document]
Comment (kira-1011) [reply scripts/pnpm-script-shell.test.js:53]: Added in b8266add: a case with `PROBE_VAR: ""` asserting the output stays empty.

File: scripts/pnpm-script-shell.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    // A caller passes `undefined` to leave the variable unset rather than empty.
+    for (const [key, value] of Object.entries(env)) if (value === undefined) delete childEnv[key];
+    const result = spawnSync("pnpm", ["run", "--silent", "probe"], {
+      cwd: dir,
+      env: childEnv,
+      encoding: "utf8",
+      shell: process.platform === "win32",
+    });
+    return { status: result.status, stdout: result.stdout.trim(), stderr: result.stderr };
+  } finally {
```
[the remark this answers is not in the document]
Comment (kira-1011) [reply scripts/pnpm-script-shell.test.js:36]: Added in b8266add: a spawn error or null output now throws with the cause instead of a TypeError.

File: scripts/pnpm-script-shell.test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  try {
+    fs.copyFileSync(path.join(repoRoot, ".npmrc"), path.join(dir, ".npmrc"));
+    fs.writeFileSync(
+      path.join(dir, "package.json"),
+      JSON.stringify({ name: "script-shell-probe", private: true, scripts: { probe: script } }),
+    );
+    const childEnv = { ...process.env, ...env };
+    // A caller passes `undefined` to leave the variable unset rather than empty.
+    for (const [key, value] of Object.entries(env)) if (value === undefined) delete childEnv[key];
+    const result = spawnSync("pnpm", ["run", "--silent", "probe"], {
```
[the remark this answers is not in the document]
Comment (kira-1011) [reply scripts/pnpm-script-shell.test.js:26]: The three behavioural cases already run a real script through pnpm with this `.npmrc` and assert the effective result, which is stronger than reading the setting back; they all fail with the file removed.

**Review by Kinfe123 — APPROVED** [review]
